functional-tests/src/test/java/com/redhat/che/functional/tests/simpleTestSuite.xml

```diff
@@ -0,0 +1,49 @@
+<!--
+
+    Copyright (c) 2016-2018 Red Hat, Inc.
+    All rights reserved. This program and the accompanying materials
+    are made available under the terms of the Eclipse Public License v1.0
+    which accompanies this distribution, and is available at
+    http://www.eclipse.org/legal/epl-v10.html
+
+    Contributors:
+      Red Hat, Inc. - initial API and implementation
+
+-->
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
+<suite name="Simple test suite"
+       verbose="10"
+       parallel="none"
+       thread-count="1">
+
+    <test name="all">
+        <method-selectors>
+            <method-selector>
+                <script language="beanshell">
+                    <![CDATA[
+    boolean isTestGroupsAgreedWithRequestedGroups() {
```

Comment:
Can you elaborate why we need to hardcode real code in XML?

Reply:
I was asking myself the same question. And after thinking this through, I think we should keep this part in here.

In upstream, they have the same code in their test suites. This code ensures, that no testcases annotated with `@Test(groups="<someTestGroup>")` are executed. 

As we want to use upstream test classes to run tests against rh-che, this mechanism can come in handy to exclude some groups of testcases from our suite.

Reply:
Thank you for the explanation, but the approach is awful. We shouldn't have such an approach upstream.

Reply:
Ok so should I remove it or leave it here until some better approach is found? As @rhopp said, this functionality can be handy.

Reply:
I'll try to reach to upstream Che QE today (or tomorrow) and discuss this one and see if we can find another (nicer) solution ;-)

Reply:
After discussing with upstream che-qe I would go with removing this code from here as we don't `need` it right now and upstream is thinking about changing that too, so hopefully the new solution will be in place, when we need it.
Che-qe team is having some troubles with this approach also and they promised to invite us to the meeting, where we could discuss this and try to solve
> iokhrimenko: We have planned to discuss our groups approach with the Dmytro at Monday because we have several problems with this functionality, and I created some temporary workaround. So if you wish I can invite you to our planning and you will take part in discussion

Reply:
Removal of *beanshell* script from the test suite xml breaks exclusion of tests which are not relevant to current Che infrastructure. At the same time that script doesn't allow to define test dedicated to the several Che infrastructure (not all) at a time (for example, just for _OpenShift_ and _OSIO_).
I have created an issue https://github.com/eclipse/che/issues/10583 to fix this problem.